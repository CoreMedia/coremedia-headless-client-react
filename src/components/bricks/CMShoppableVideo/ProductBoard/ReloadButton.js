// @flow
import React from 'react';
import PropTypes from 'prop-types';

import { IconButton as BasicIconButton } from '../../../basic/Button';
import RepeatIcon from '../../../basic/Icon/RepeatIcon';
import { media } from '../../../../styles/themes/utils';

const IconButton = BasicIconButton.extend`
  width: 100%;
  height: auto;
  ${props => media(props.theme.breakpoints.tablet)`
    width: 40px;
    height: 40px;
  `};
`;

IconButton.displayName = 'IconButton';

IconButton.propTypes = {
  theme: PropTypes.shape({
    breakpoints: PropTypes.shape({
      tablet: PropTypes.number.isRequired,
    }).isRequired,
  }),
};

IconButton.defaultProps = {
  theme: {
    breakpoints: {
      tablet: 768,
    },
  },
};

type Props = {
  id?: string,
  handleClick: (ev: SyntheticEvent<HTMLButtonElement>) => void,
};

const ReloadButton = ({ id, handleClick }: Props) => (
  <IconButton onClick={handleClick} color="#808080" backgroundColor="rgba(255, 255, 255, 0.5)">
    <RepeatIcon id={id} />
  </IconButton>
);

ReloadButton.propTypes = {
  id: PropTypes.string,
  handleClick: PropTypes.func.isRequired,
};

export default ReloadButton;
