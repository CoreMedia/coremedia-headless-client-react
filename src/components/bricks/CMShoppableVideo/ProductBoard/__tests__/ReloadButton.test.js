// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import ReloadButton from '../ReloadButton';

describe('ReloadButton Component', () => {
  it('should render correctly', () => {
    const handleClick = jest.fn().mockName('handleClick');
    const tree = renderer.create(<ReloadButton id="id123" handleClick={handleClick} />);
    expect(tree).toMatchSnapshot();
  });
  it('should call handleClick', () => {
    const handleClick = jest.fn().mockName('handleClick');
    const wrapper = shallow(<ReloadButton id="id123" handleClick={handleClick} />);
    wrapper.simulate('click');
    expect(handleClick).toHaveBeenCalled();
  });
});
