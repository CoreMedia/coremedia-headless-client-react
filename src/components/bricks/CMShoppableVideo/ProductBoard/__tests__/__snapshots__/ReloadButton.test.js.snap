// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReloadButton Component should render correctly 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  width: 100%;
  height: 100%;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 40px;
  height: 40px;
  padding: 7px;
  margin: 0;
  border-style: none;
  border-radius: 50%;
  border-width: 0;
  outline: none;
  color: #808080;
  background-color: rgba(255,255,255,0.5);
  cursor: pointer;
  width: 100%;
  height: auto;
}

@media (min-width:48em) {
  .c0 {
    width: 40px;
    height: 40px;
  }
}

<button
  className="c0"
  color="#808080"
  onClick={[MockFunction handleClick]}
>
  <svg
    aria-labelledby="ID"
    className="c1"
    role="img"
    size={undefined}
    viewBox="0 0 84 92"
  >
    <title
      id="ID"
    >
      repeat
    </title>
    <g
      fill="none"
      fillRule="evenodd"
      stroke="currentColor"
    >
      <path
        d="M3.477 66.46c10.67 20.066 35.586 27.683 55.652 17.014 20.066-10.67 27.684-35.586 17.014-55.652C65.473 7.756 40.558.138 20.492 10.808a40.992 40.992 0 0 0-13.626 11.66"
        strokeWidth="6"
      />
      <path
        d="M23.648 24.024H6.385V6.762z"
        fill="currentColor"
        strokeWidth="5"
      />
    </g>
  </svg>
</button>
`;
